doc/spinel-protocol-src/spinel-tech-thread.md

```diff
@@ -249,12 +249,12 @@ The stable leader network data.
 ### PROP 5391: PROP_THREAD_JOINERS {#prop-thread-joiners}
 
 * Type: Insert/Remove Only (optionally Read-Write)
-* Packed-Encoding: `A(t(ULE))`
+* Packed-Encoding: `A(t(dLE))`
```

Comment:
Please justify why this change is necessary or desirable.

Reply:
As per Thread spec, PSKd is a binary object, which can contain 0x00 bytes, i.e. null-characters. So far only ASCII strings were used as PSKds. This change is supposed to allow for non-ASCII PSKds.

Reply:
The Joining Device Credential is encoded using characters.  The PSKd is encoded in binary.

From the Thread 1.1.1 specification:

> Joining Device Credential: A human-scaled passphrase for use in authenticating that a new Joiner device is the correct one. ... This credential when specifically encoded in binary is referred to as the PSKd (Pre- Shared Key for the Device). A Joining Device Credential is encoded as uppercase alphanumeric characters (base32-thread: 0-9,A-Y excluding I,O,Q, and Z for readability) with a minimum length of 6 such characters and a maximum length of 32 such characters.

Reply:
Is the PSKd just a decoded Joining Device Credential?

Reply:
Yes.

Reply:
Then why not make this field the Joining Device Credential field?

Reply:
It seems like the obvious thing to do, especially if the base32 encoding hasn't even been finalized yet.

Reply:
Wouldn't it be easier to re-define the field to be the Joining Device Credential instead of the raw PSKd? I mean, the whole reason it was originally defined as a string in Spinel was because it was intended to be the "Joining Device Credential"